instrumentation/servlet/src/main/java/io/opentelemetry/auto/instrumentation/servlet/dispatcher/RequestDispatcherInstrumentation.java

```diff
@@ -125,28 +129,31 @@ public static SpanWithScope start(
 
       // save the original servlet span before overwriting the request attribute, so that it can be
       // restored on method exit
-      originalServletSpan = request.getAttribute(SPAN_ATTRIBUTE);
+      originalServletContext = request.getAttribute(CONTEXT_ATTRIBUTE);
 
       // this tells the dispatched servlet to use the current span as the parent for its work
-      request.setAttribute(SPAN_ATTRIBUTE, span);
+      Context newContext = withSpan(span, Context.current());
+      request.setAttribute(CONTEXT_ATTRIBUTE, newContext);
 
-      return new SpanWithScope(span, currentContextWith(span));
+      return new SpanWithScope(span, withScopedContext(newContext));
     }
 
     @Advice.OnMethodExit(onThrowable = Throwable.class, suppress = Throwable.class)
     public static void stop(
         @Advice.Enter final SpanWithScope spanWithScope,
-        @Advice.Local("_originalServletSpan") final Object originalServletSpan,
+        @Advice.Local("_originalServletContext") final Object originalServletContext,
         @Advice.Argument(0) final ServletRequest request,
         @Advice.Thrown final Throwable throwable) {
       if (spanWithScope == null) {
         return;
       }
 
       // restore the original servlet span
-      // since spanWithScope is non-null here, originalServletSpan must have been set with the prior
+      // since spanWithScope is non-null here, originalServletContext must have been set with the
+      // prior
       // servlet span (as opposed to remaining unset)
-      request.setAttribute(SPAN_ATTRIBUTE, originalServletSpan);
+      // TODO review this logic. Seems like manual context management
```

Comment:
hm, good question, i think `include` and `forward` are synchronous, so thread-local context propagation should be enough

Reply:
Made #579 to come back to this later.